docs/api/conventions.md

```diff
@@ -91,29 +91,147 @@ There are a few conventions that Remix uses you should be aware of.
 - **`app/entry.client.{js,jsx,ts,tsx}`**: This is your entry into the browser rendering/hydration piece of Remix. This file is required.
 
-### Route Filenames
+### Route File Conventions
 
-- **`app/root.tsx`**: This is your root layout, or "root route" (very sorry for those of you who pronounce those words the same way!). It works just like all other routes: you can export a `loader`, `action`, etc.
-- **`app/routes/*.{js,jsx,ts,tsx,md,mdx}`**: Any files in the `app/routes/` directory will become routes in your application. Remix supports all of those extensions.
-- **`app/routes/{folder}/*.tsx`**: Folders inside of routes will create nested URLs.
-- **`app/routes/{folder}` with `app/routes/{folder}.tsx`**: When a route has the same name as a folder, it becomes a "layout route" for the child routes inside the folder. Render an `<Outlet />` and the child routes will appear there. This is how you can have multiple levels of persistent layout nesting associated with URLs.
-- **Dots in route filenames**: Adding a `.` in a route file will create a nested URL, but not a nested layout. Flat files are flat layouts, nested files are nested layouts. The `.` allows you to create nested URLs without needing to create a bunch of layouts. For example: `app/routes/some.long.url.tsx` will create the URL `/some/long/url`.
-- **`app/routes/index.tsx`**: Routes named "index" will render when the parent layout route's path is matched exactly.
-- **`$param`**: The dollar sign denotes a dynamic segment of the URL. It will be parsed and passed to your loaders and routes.
+Setting up routes in Remix is as simple as creating files in your `app` directory. These are the conventions you should know to understand how routing in Remix works.
 
-  For example: `app/routes/users/$userId.tsx` will match the following URLs: `users/123` and `users/abc` but not `users/123/abc` because that has too many segments. See the <Link to="../routing">routing guide</Link> for more information.
+Please note that you can use either `.jsx` or `.tsx` file extensions depending on whether or not you use TypeScript. We'll stick with `.tsx` in the examples to avoid duplication (and because we ❤️ TypeScript).
 
-  Some CLIs require you to escape the \$ when creating files:
+#### Root Layout Route
 
-  ```bash
-  touch routes/\$params.tsx
-  ```
+```[3]
+app
+| routes
+| root.tsx
+```
+
+This is your root layout, or "root route" (very sorry for those of you who pronounce those words the same way!). It works just like all other routes:
+
+- You can export a [`loader`](#loader), [`action`](#action), [`meta`](#meta), [`headers`](#headers), or [`links`](#links) function
+- You can export an [`ErrorBoundary`](#errorboundary) or [`CatchBoundary`](#catchboundary)
+- Your default export is the layout component that renders the rest of your app in an [`<Outlet />`](https://reactrouter.com/docs/en/v6/api#outlet)
+
+#### Basic Routes
+
+```[3-4]
+app
+| routes
+| - about.tsx
+| - index.tsx
+| root.tsx
+```
```

Comment:
For each of these, could we get another block that shows the URLs this would support? Or maybe even the output of `remix routes`? Could put that in a `<details>` if it's too big. I think that would help a great deal.